torchmetrics/text/eed.py

```diff
@@ -0,0 +1,143 @@
+        >>> metric = EED()
+        >>> metric(reference_corpus=reference_corpus, hypothesis_corpus=hypothesis_corpus)
+        tensor(0.3204)
+
+    References:
+        [1] P. Stanchev, W. Wang, and H. Ney, “EED: Extended Edit Distance Measure for Machine Translation”, submitted
+        to WMT 2019. `EED`_
+    """
+
+    scores: Tensor
+    total_num_sentences: Tensor
+    higher_is_better: False
+    is_differentiable: False
+
+    def __init__(
+        self,
+        language: Literal["en", "ja"] = "en",
+        return_sentence_level_score: bool = False,
+        compute_on_step: bool = True,
+        dist_sync_on_step: bool = False,
+        process_group: Optional[Any] = None,
+        dist_sync_fn: Callable = None,
+        alpha: float = 2.0,
+        rho: float = 0.3,
+        deletion: float = 0.2,
+        insertion: float = 1.0,
+    ):
+        super().__init__(
+            compute_on_step=compute_on_step,
+            dist_sync_on_step=dist_sync_on_step,
+            process_group=process_group,
+            dist_sync_fn=dist_sync_fn,
+        )
+
+        if language not in ("en", "ja"):
+            raise ValueError(f"Expected argument `language` to either be `en` or `ja` but got {language}")
+        self.language: Literal["en", "ja"] = language
+        self.return_sentence_level_score = return_sentence_level_score
+        self.sentence_eed = []
+
+        self.alpha = alpha
+        self.rho = rho
+        self.deletion = deletion
+        self.insertion = insertion
+
+        self.add_state("scores", tensor(0.0), dist_reduce_fx="sum")
+        self.add_state("total_num_sentences", tensor(0.0), dist_reduce_fx="sum")
+        if self.return_sentence_level_score:
+            self.add_state("sentence_eed", [], dist_reduce_fx="cat")
+
+    def update(  # type: ignore
+        self,
+        reference_corpus: Sequence[Union[str, Sequence[str]]],
+        hypothesis_corpus: Union[str, Sequence[str]],
+    ) -> None:
+        """Update EED statistics.
+
+        Args:
+            reference_corpus: An iterable of iterables of reference corpus
+            hypothesis_corpus: An iterable of hypothesis corpus
+        """
+        scores, total_num_sentences, sentence_eed = _eed_update(
+            reference_corpus,
+            hypothesis_corpus,
+            self.language,
+            self.return_sentence_level_score,
+            self.alpha,
+            self.rho,
+            self.deletion,
+            self.insertion,
+        )
+
+        self.scores += scores
+        self.total_num_sentences += total_num_sentences
+        if self.return_sentence_level_score is True:
+            self.sentence_eed.extend(sentence_eed)
+
+    def compute(self) -> Tensor:
+        """Calculate extended edit distance score.
+
+        Returns:
+            Extended edit distance score as tensor
+        """
+        eed = _eed_compute(self.scores, self.total_num_sentences)
+        return eed
```

Comment:
We need to allow to return sentence-level scores

Reply:
adding on next commit